tests/migration/sle12_online_migration/yast2_migration.pm

```diff
@@ -204,9 +204,7 @@ sub run {
     assert_screen 'yast2-migration-target';
     send_key "alt-p";                                       # focus on the item of possible migration targets
     send_key_until_needlematch 'migration-target-' . get_var("VERSION"), 'down', 5;
-    wait_still_screen 5;
-    send_key "alt-n";
-    assert_screen ['yast2-migration-installupdate', 'yast2-migration-proposal'], 500;
+    send_key_until_needlematch ['yast2-migration-installupdate', 'yast2-migration-proposal'], 'alt-n';
```

Comment:
As you asked me for advice I looked into the issue in more detail and I found one of the common problems: Most likely there is no real issue with sending key presses but that the prerequisites are not properly fulfilled and it already starts in L204 . See https://openqa.suse.de/tests/3760125#step/yast2_migration/18 . The needle does not assert that the "Next" key is shown properly, hence the system might not yet be fully ready. Then afterwards similar problematic needles are used. The needle "yast2_migration-yast2-migration-installupdate-20190923" in http://openqa.nue.suse.com/tests/3946437#step/yast2_migration/37 is wrong. It does not assert the system to be in a ready state to process any next step. It must be deleted and replaced by a proper one. For reference take a look in https://openqa.opensuse.org/tests/1193215#step/installation_overview/1 . The "inst-overview" needle not only looks for text within the text box but also that the "install" button is ready. Also in https://openqa.suse.de/tests/3760125#step/yast2_migration/21 the migration target is checked but not that the page is fully loaded as the description text is not yet shown. This can also explain why a system does not react to key presses yet.

Reply:
I see that you updated your code but kept the `send_key_until_needlematch` but haven't responded to my comment so I am not sure if you have seen it. IMHO the repeated key presses can cause problems but are still the right approach *if* even after ensuring proper needles you encountered problems.

Reply:
Hi Oliver, I'm struggling in the issue for send_key_until_needlematch these days and I haven't respond you in time, sorry.

In fact I found the if we use send_key_until_needlematch, it will send key after needle matched, I think it is the issue of the function. And I tried to fix it using the parameter of itself such as set timeout value but failed, so you can see I have rebase it several times.
Please see below link for the issue, it matched yast2-migration-proposal but then the page changed to next page that means it send next key after matched yast2-migration-proposal.
https://openqa.nue.suse.com/tests/3949902#step/yast2_migration/38

Another run, http://openqa.nue.suse.com/tests/3952678#step/yast2_migration/31 . It seems the key is wrong for yast2-migration-installupdate, I guess the test code hasn't been run here, it should be alt-u while not alt-y. I suppose in previous run it will match the 'yast2-migration-proposal' and not the ''yast2-migration-proposal'. 

While I ready carefully for your comments, I found I missed to remove https://openqa.suse.de/tests/3760125#step/yast2_migration/18. I just remove it by a MR and create a new needle with the next button. I have run three verify runs there only with the old code, I think maybe it can work. If it works, I think we can close this PR and file a new ticket for the further issue of here, such as what I describe in previous paragraph.  If not, we need fix the send more key issue for send_key_until_needlematch.
https://openqa.nue.suse.com/tests/3953803#settings
https://openqa.nue.suse.com/tests/3953804#settings
http://openqa.nue.suse.com/tests/3953853#settings